pymc3/distributions/mixture.py

```diff
@@ -612,3 +618,183 @@ def __init__(self, w, mu, sigma=None, tau=None, sd=None, comp_shape=(), *args, *
 
     def _distr_parameters_for_repr(self):
         return ["w", "mu", "sigma"]
+
+
+class MixtureSameFamily(Distribution):
+    R"""
+    Mixture log-likelihood
+
+    .. math::f(x \mid w, \theta) = \sum_{i = 1}^n w_i f_i(x \mid \theta_i)\textrm{ Along mixture\_axis}
+
+    ========  ============================================
+    Support   :math:`\textrm{support}(f)`
+    Mean      :math:`w\mu`
+    ========  ============================================
+
+    Parameters
+    ----------
+    w: array of floats
+        w >= 0 and w <= 1
+        the mixture weights
+    comp_dists: multidimensional PyMC3 distribution (e.g. `pm.Multinomial.dist(...)`)
+        with shape (batch_shape, mixture_axis, event_shape)
+    mixture_axis: int
+        Axis to be reduced in the mixture
+    """
+
+    def __init__(self, w, comp_dists, mixture_axis=-1, *args, **kwargs):
+        self.w = tt.as_tensor_variable(w)
+        if not isinstance(comp_dists, Distribution):
+            raise TypeError(
+                "The MixtureSameFamily distribution only accepts Distribution "
+                f"instances as its components. Got {type(comp_dists)} instead."
+            )
+        self.comp_dists = comp_dists
+        if mixture_axis < 0:
+            mixture_axis = len(comp_dists.shape) + mixture_axis
+        comp_shape = to_tuple(comp_dists.shape)
+        self.shape = comp_shape[:mixture_axis] + comp_shape[mixture_axis + 1 :]
+        self.mixture_axis = mixture_axis
+        kwargs.setdefault("dtype", self.comp_dists.dtype)
+
+        # Computvalte the mode so we don't always have to pass a tes
```

Comment:
I'm guessing this sentence is:
```suggestion
        # Compute the mode so we don't always have to pass a test:
```

Reply:
It actually was supposed to say `# Compute the mode so we don't always have to pass a testval` but the mouse cursor seems to have betrayed us.